# Рынок видеоигр

- Автор: Айриян Луиза
- Дата:07.03.2026

### Цели и задачи проекта

- Отобрать данные по времени выхода игры.
- Категоризовать игры по оценкам пользователей и экспертов.

### Описание данных
В проекте будут использованы данные датасета `/datasets/new_games.csv` со следующим описанием:
- `Name` — название игры.
- `Platform` — название платформы.
- `Year of Release` — год выпуска игры.
- `Genre`— жанр игры.
- `NA sales` — продажи в Северной Америке (в миллионах проданных копий).
- `EU sales` — продажи в Европе (в миллионах проданных копий).
- `JP sales` — продажи в Японии (в миллионах проданных копий).
- `Other sales` — продажи в других странах (в миллионах проданных копий).
- `Critic Score` — оценка критиков (от 0 до 100).
- `User Score` — оценка пользователей (от 0 до 10).
- `Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

### Содержимое проекта
 1. *Загрузка данных и знакомство с ними*
 2. *Проверка ошибок в данных и их предобработка*
 3. *Фильтрация данных*
 4. *Категоризация данных*
 5. *Итоговый вывод*

## 1. Загрузка данных и знакомство с ними

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


In [1]:
import pandas as pd

In [2]:
df_news = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

In [3]:
df_news.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [4]:
df_news.head()

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


- Сделайте вывод о полученных данных: данные какого объёма вам предоставили, соответствуют ли они описанию, встречаются ли в них пропуски, используются ли верные типы данных.
- Отметьте другие особенности данных, которые вы обнаружили и на которые стоит обратить внимание при предобработке. Например, вы можете проверить названия столбцов: все ли названия отражают содержимое данных и прописаны в удобном для работы виде.

Датасет `/datasets/new_games.csv` содержит 11 столбцов и 16956 строк, в которых представлена информация о видеоиграх.

Изучим типы данных и их корректность:
- **Числовые значения с плавающей запятой (float64).** Один столбец, `price for the night`, содержит стоимость за ночь и представлен типом `float64`. Это верное решение, так как цена может включать дробные значения.
- **Строковые данные (object).** Семь столбцов имеют тип данных `object`:
    - ` Name`, `Platform` и `Genre` содержат строковую информацию (название игры, платформа и  жанр игры), что логично для текстовых данных. Здесь тип данных `object` подходит.
    - `   JP sales  `, `EU sales` `NA sales` и  `Other sales` хранят информацию о продажах игр в других странах. 
    


## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [5]:
# Выведим все названия столбцов датафрейма и проверим их стиль написания 
for column in df_news.columns:
    display(column)

'Name'

'Platform'

'Year of Release'

'Genre'

'NA sales'

'EU sales'

'JP sales'

'Other sales'

'Critic Score'

'User Score'

'Rating'

In [6]:
# Приводим столбцы к стилю  snake case 
df_news = df_news.copy() #сохраним оригинальные данные 
df_news.columns = (df_news.columns.str.lower().str.replace(' ', '_'))

In [7]:
# Выведем переименованные названия столбцов датафрейма 
for column in df_news.columns:
    display(column)


'name'

'platform'

'year_of_release'

'genre'

'na_sales'

'eu_sales'

'jp_sales'

'other_sales'

'critic_score'

'user_score'

'rating'

### 2.2. Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

In [8]:
# 1. Обработка столбца year_of_release
# Заполняем пропуски медианным значением
median_year = df_news['year_of_release'].median()
df_news['year_of_release'] = df_news['year_of_release'].fillna(median_year)


In [9]:
# Преобразуем в int64
df_news['year_of_release'] = df_news['year_of_release'].astype('int64')

In [10]:
# 2. Обработка столбцов Sales
sales_columns = ['na_sales', 'eu_sales', 'jp_sales', 'other_sales']
for col in sales_columns:
    # Заменяем нечисловые значения на NaN и преобразуем в float
    df_news[col] = pd.to_numeric(df_news[col], errors='coerce')
    # Заполняем пропуски нулями (отсутствие продаж)
    df_news[col] = df_news[col].fillna(0.0)


In [11]:
# 3. Обработка critic_score
# Преобразуем в числовой тип, заменяя нечисловые значения на NaN
df_news['critic_score'] = pd.to_numeric(df_news['critic_score'], errors='coerce')
# Заполняем пропуски средним значением
mean_critic = df_news['critic_score'].mean()
df_news['critic_score'] = df_news['critic_score'].fillna(mean_critic)

In [12]:
# 4. Обработка user_score
# Заменяем 'tbd' на NaN
df_news['user_score'] = df_news['user_score'].replace('tbd', 'pd.NaN')
# Преобразуем в числовой тип
df_news['user_score'] = pd.to_numeric(df_news['user_score'], errors='coerce')
# Заполняем пропуски средним значением
mean_user = df_news['user_score'].mean()
df_news['user_score'] = df_news['user_score'].fillna(mean_user)

In [13]:
# 5. Проверка текстовых столбцов
text_columns = ['name', 'platform', 'genre', 'rating']
for col in text_columns:
    df_news[col] = df_news[col].astype('object')  # Приводим к типу object (текст)


In [14]:
def convert_to_numeric(column, string_values_to_replace=None):
    if string_values_to_replace is None:
        string_values_to_replace = ['unknown', 'Unknown', 'UNKNOWN',
                           'n/a', 'N/A', 'na', '-', '.', '']
    column_cleaned = column.replace(string_values_to_replace)
    column_numeric = pd.to_numeric(column_cleaned, errors='coerce')
    return column_numeric

In [15]:
# Список числовых столбцов
numeric_columns = [
    'year_of_release',
    'na_sales', 'eu_sales', 'jp_sales', 'other_sales',
    'critic_score', 'user_score'
]

In [16]:
# Преобразование всех числовых столбцов
for col in numeric_columns:
    df_news[col] = convert_to_numeric(df_news[col])

### 2.3. Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [17]:
# Загрузка данных
df_news = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

In [18]:
# Подсчёт пропусков по столбцам
nan_counts = df_news.isna().sum()
nan_percentage = (df_news.isna().sum() / len(df_news)) * 100

In [19]:
# Объединение результатов в таблицу
missing_data = pd.concat([nan_counts, nan_percentage], axis=1)
missing_data.columns = ['Количество пропусков', 'Процент пропусков']
missing_data = missing_data[missing_data['Количество пропусков'] > 0]
display(missing_data)

,Количество пропусков,Процент пропусков
Name,2,0.011795
Year of Release,275,1.621845
Genre,2,0.011795
Critic Score,8714,51.391838
User Score,6804,40.127389
Rating,6871,40.522529


**В данных обнаружены пропуски в шести столбцах: `Name` (2), `Genre` (2) , `Year of Release` (275),`User Score` (6804), `Rating` (6871), `Critic Score` (8714). Наибольшее количество пропусков приходится на последний столбец.**

### 2.4. Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [20]:
# Загрузка данных
df_news = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

# Анализ уникальных значений
categorical_columns = ['Genre', 'Platform', 'Rating', 'Year of Release']

for col in categorical_columns:
    display(f"\n--- {col} ---")
    unique_values = df_news[col].unique()
    display(f"Количество уникальных значений: {len(unique_values)}")
    

'\n--- Genre ---'

'Количество уникальных значений: 25'

'\n--- Platform ---'

'Количество уникальных значений: 31'

'\n--- Rating ---'

'Количество уникальных значений: 9'

'\n--- Year of Release ---'

'Количество уникальных значений: 38'

In [21]:
# Выводим первые 20 уникальных значений 
display('Уникальные значения:')
for val in unique_values[:20]:
        display(f" - {val}")

'Уникальные значения:'

' - 2006.0'

' - 1985.0'

' - 2008.0'

' - 2009.0'

' - 1996.0'

' - 1989.0'

' - 1984.0'

' - 2005.0'

' - 1999.0'

' - 2007.0'

' - 2010.0'

' - 2013.0'

' - 2004.0'

' - 1990.0'

' - 1988.0'

' - 2002.0'

' - 2001.0'

' - 2011.0'

' - 1998.0'

' - 2015.0'

In [22]:
# Нормализация регистра
df_news['Genre'] = df_news['Genre'].str.lower()
df_news['Platform'] = df_news['Platform'].str.upper()
df_news['Rating'] = df_news['Rating'].str.upper()


In [23]:
# Обработка Year of Release: приводим к целому числу
df_news['Year of Release'] = pd.to_numeric(df_news['Year of Release'], errors='coerce')

In [24]:
for col in categorical_columns:
    display(f"\n--- {col} ---")
    unique_after = df_news[col].dropna().unique()  # исключаем NaN для подсчёта
    display(f"Количество уникальных значений после: {len(unique_after)}")
    
    # Показываем первые 20 значений после нормализации
    display("Уникальные значения после :")
    for val in sorted(unique_after)[:20]:  # сортировка для удобства сравнения
       display(f"  - {val}")

'\n--- Genre ---'

'Количество уникальных значений после: 12'

'Уникальные значения после :'

'  - action'

'  - adventure'

'  - fighting'

'  - misc'

'  - platform'

'  - puzzle'

'  - racing'

'  - role-playing'

'  - shooter'

'  - simulation'

'  - sports'

'  - strategy'

'\n--- Platform ---'

'Количество уникальных значений после: 31'

'Уникальные значения после :'

'  - 2600'

'  - 3DO'

'  - 3DS'

'  - DC'

'  - DS'

'  - GB'

'  - GBA'

'  - GC'

'  - GEN'

'  - GG'

'  - N64'

'  - NES'

'  - NG'

'  - PC'

'  - PCFX'

'  - PS'

'  - PS2'

'  - PS3'

'  - PS4'

'  - PSP'

'\n--- Rating ---'

'Количество уникальных значений после: 8'

'Уникальные значения после :'

'  - AO'

'  - E'

'  - E10+'

'  - EC'

'  - K-A'

'  - M'

'  - RP'

'  - T'

'\n--- Year of Release ---'

'Количество уникальных значений после: 37'

'Уникальные значения после :'

'  - 1980.0'

'  - 1981.0'

'  - 1982.0'

'  - 1983.0'

'  - 1984.0'

'  - 1985.0'

'  - 1986.0'

'  - 1987.0'

'  - 1988.0'

'  - 1989.0'

'  - 1990.0'

'  - 1991.0'

'  - 1992.0'

'  - 1993.0'

'  - 1994.0'

'  - 1995.0'

'  - 1996.0'

'  - 1997.0'

'  - 1998.0'

'  - 1999.0'

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [25]:
# Нормализация данных 
df_news['Genre'] = df_news['Genre'].str.lower()
df_news['Platform'] = df_news['Platform'].str.upper()
df_news['Rating'] = df_news['Rating'].str.upper()


In [26]:
# Исправление явных опечаток
genre_mapping = {'rpg': 'role-playing', 'puzle': 'puzzle', 'shooter': 'shooting',
                  'sim': 'simulation', 'sport': 'sports'}
platform_mapping = {'PS3': 'PLAYSTATION 3', 'PS4': 'PLAYSTATION 4',
                    'X360': 'XBOX 360', 'XB1': 'XBOX ONE', 'WII': 'WII', 'NINTENDO WII': 'WII'}
rating_mapping = {'EVERYONE': 'E', 'K-A': 'E'}
df_news['Genre'] = df_news['Genre'].replace(genre_mapping)
df_news['Platform'] = df_news['Platform'].replace(platform_mapping)
df_news['Rating'] = df_news['Rating'].replace(rating_mapping)

In [27]:
# Приведение года выпуска к числовому формату
df_news['Year of Release'] = pd.to_numeric(df_news['Year of Release'], errors='coerce')

In [28]:
# Поиск полных дубликатов
full_duplicates = df_news[df_news.duplicated(keep=False)] 

display(f"Всего полных дубликатов: {len(full_duplicates)}")
display(f"Уникальных записей среди дубликатов: {full_duplicates.drop_duplicates().shape[0]}")

if not full_duplicates.empty:
    display(full_duplicates.head(10))

'Всего полных дубликатов: 482'

'Уникальных записей среди дубликатов: 241'

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
267,Batman: Arkham Asylum,PLAYSTATION 3,2009.0,action,2.24,1.31,0.07,0.61,91.0,8.9,T
268,Batman: Arkham Asylum,PLAYSTATION 3,2009.0,action,2.24,1.31,0.07,0.61,91.0,8.9,T
367,James Bond 007: Agent Under Fire,PS2,2001.0,shooting,1.90,1.13,0.1,0.41,72.0,7.9,T
368,James Bond 007: Agent Under Fire,PS2,2001.0,shooting,1.90,1.13,0.1,0.41,72.0,7.9,T
716,God of War: Ascension,PLAYSTATION 3,2013.0,action,1.23,0.63,0.04,0.35,80.0,7.5,M
717,God of War: Ascension,PLAYSTATION 3,2013.0,action,1.23,0.63,0.04,0.35,80.0,7.5,M
822,Wipeout: The Game,WII,2009.0,misc,1.94,0.0,0.0,0.12,NaN,NaN,NaN
823,Wipeout: The Game,WII,2009.0,misc,1.94,0.0,0.0,0.12,NaN,NaN,NaN
847,Rayman Raving Rabbids: TV Party,WII,2008.0,misc,0.72,1.08,0.0,0.23,73.0,7.7,E10+
848,Rayman Raving Rabbids: TV Party,WII,2008.0,misc,0.72,1.08,0.0,0.23,73.0,7.7,E10+


- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.


In [29]:
# Группируем по названию и платформе, смотрим разброс годов выпуска
grouped = df_news.groupby(['Name', 'Platform'])['Year of Release'].agg(['count', 'min', 'max', 'nunique']).reset_index()
year_discrepancies = grouped[grouped['nunique'] > 1]


In [30]:
df_cleaned = df_news.drop_duplicates()
display(f"Исходный размер: {df_news.shape[0]}, после удаления полных дубликатов: {df_cleaned.shape[0]}")

'Исходный размер: 16956, после удаления полных дубликатов: 16715'

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

- После проведения предобработки данных напишите общий промежуточный вывод.

### В ходе предварительной обработки данных выполнены следующие действия:

- удалено 241 строка
- найдены 2 записи с расхождением в годах


## 3. Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [31]:
# Нормализация данных (если ещё не выполнена)
df_news['Genre'] = df_news['Genre'].str.lower()
df_news['Platform'] = df_news['Platform'].str.upper()
df_news['Rating'] = df_news['Rating'].str.upper()

# Исправление явных опечаток (пример)
genre_mapping = {'rpg': 'role-playing', 'puzle': 'puzzle', 'shooter': 'shooting',
                  'sim': 'simulation', 'sport': 'sports'}
platform_mapping = {'PS3': 'PLAYSTATION 3', 'PS4': 'PLAYSTATION 4',
                    'X360': 'XBOX 360', 'XB1': 'XBOX ONE', 'WII': 'WII', 'NINTENDO WII': 'WII'}
rating_mapping = {'EVERYONE': 'E', 'K-A': 'E'}

df_news['Genre'] = df_news['Genre'].replace(genre_mapping)
df_news['Platform'] = df_news['Platform'].replace(platform_mapping)
df_news['Rating'] = df_news['Rating'].replace(rating_mapping)

# Приведение года выпуска к числовому формату
df_news['Year of Release'] = pd.to_numeric(df_news['Year of Release'], errors='coerce')

# Отбор данных за 2000–2013 гг. включительно
df_actual = df_news[(df_news['Year of Release'] >= 2000) & (df_news['Year of Release'] <= 2013)].copy()

# Сортировка по году выпуска для удобства анализа
df_actual = df_actual.sort_values('Year of Release')

In [32]:
display("=== СТАТИСТИКА ПО ПЕРИОДУ 2000–2013 ГГ. ===")
display(f"Количество записей: {len(df_actual)}")


# Распределение по годам
year_distribution = df_actual['Year of Release'].value_counts().sort_index()
display(year_distribution)

# Уникальные значения ключевых категорий
display(f"\nУникальных платформ: {df_actual['Platform'].nunique()}")
display(f"Уникальных жанров: {df_actual['Genre'].nunique()}")
display(f"Уникальных рейтингов: {df_actual['Rating'].nunique()}")


'=== СТАТИСТИКА ПО ПЕРИОДУ 2000–2013\xa0ГГ. ==='

'Количество записей: 12980'

Year of Release
2000.0     357
2001.0     491
2002.0     839
2003.0     789
2004.0     771
2005.0     950
2006.0    1020
2007.0    1218
2008.0    1445
2009.0    1450
2010.0    1279
2011.0    1149
2012.0     670
2013.0     552
Name: count, dtype: int64

'\nУникальных платформ: 20'

'Уникальных жанров: 12'

'Уникальных рейтингов: 7'

## 4. Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [33]:
# Убедимся, что User Score имеет числовой тип
df_actual['User Score'] = pd.to_numeric(df_actual['User Score'], errors='coerce')

# Создаём функцию для категоризации оценок
def categorize_user_score(score):
    if pd.isna(score):  # Обработка пропусков
        return 'Unknown'
    elif 8 <= score <= 10:
        return 'Высокая оценка'
    elif 3 <= score < 8:
        return 'Средняя оценка'
    elif 0 <= score < 3:
        return 'Низкая оценка'
    else:
        return 'Некорректное значение'  # аномалий


In [34]:
# Применяем функцию к столбцу User Score
df_actual['User Score Category'] = df_actual['User Score'].apply(categorize_user_score)

In [43]:
print("=== РАСПРЕДЕЛЕНИЕ ИГР ПО ОЦЕНКАМ ПОЛЬЗОВАТЕЛЕЙ ===")
category_counts = df_actual['User Score Category'].value_counts()
category_percentages = df_actual['User Score Category'].value_counts(normalize=True) * 100

for category in ['Высокая оценка', 'Средняя оценка', 'Низкая оценка']:
    count = category_counts.get(category, 0)
    percent = category_percentages.get(category, 0.0)
    print(f"{category}: {count} игр ({percent:.2f}%)")

# Общая статистика
total_games = len(df_actual)
total_with_scores = total_games - category_counts.get('Unknown', 0)
print(f"\nВсего игр: {total_games}")
print(f"Из них с оценками пользователей: {total_with_scores} ({(total_with_scores / total_games * 100):.2f}%)")


=== РАСПРЕДЕЛЕНИЕ ИГР ПО ОЦЕНКАМ ПОЛЬЗОВАТЕЛЕЙ ===
Высокая оценка: 2307 игр (17.77%)
Средняя оценка: 4148 игр (31.96%)
Низкая оценка: 117 игр (0.90%)

Всего игр: 12980
Из них с оценками пользователей: 6572 (50.63%)


- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [44]:
# Создаём функцию для категоризации оценок критиков
def categorize_critic_score(score):
    if pd.isna(score):  # Обработка пропусков
        return 'Unknown'
    elif 80 <= score <= 100:
        return 'Высокая оценка'
    elif 30 <= score < 80:
        return 'Средняя оценка'
    elif 0 <= score < 30:
        return 'Низкая оценка'
    else:
        return 'Некорректное значение'  # анамалии 



In [37]:
# Применяем функцию к столбцу Critic Score
df_actual['Critic Score Category'] = df_actual['Critic Score'].apply(categorize_critic_score)


In [45]:
print("=== РАСПРЕДЕЛЕНИЕ ИГР ПО ОЦЕНКАМ КРИТИКОВ ===")
category_counts = df_actual['Critic Score Category'].value_counts()
category_percentages = df_actual['Critic Score Category'].value_counts(normalize=True) * 100


for category in ['Высокая оценка', 'Средняя оценка', 'Низкая оценка', ]:
    count = category_counts.get(category, 0)
    percent = category_percentages.get(category, 0.0)
    print(f"{category}: {count} игр ({percent:.2f}%)")


=== РАСПРЕДЕЛЕНИЕ ИГР ПО ОЦЕНКАМ КРИТИКОВ ===
Высокая оценка: 1712 игр (13.19%)
Средняя оценка: 5500 игр (42.37%)
Низкая оценка: 55 игр (0.42%)


- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [46]:
# Группируем по категориям и считаем количество игр
category_counts = df_actual['Critic Score Category'].value_counts()
category_percentages = df_actual['Critic Score Category'].value_counts(normalize=True) * 100

In [47]:
# Итоговая статистика
total_games = len(df_actual)
games_with_scores = total_games - category_counts.get('Unknown', 0)
valid_games = games_with_scores - category_counts.get('Некорректное значение', 0)

print(f"\n--- ИТОГОВАЯ СТАТИСТИКА ---")
print(f"Всего игр в датафрейме: {total_games}")
print(f"Игр с оценками критиков: {games_with_scores} ({(games_with_scores / total_games * 100):.2f}%)")
print(f"Игр с корректными оценками (0–100): {valid_games} ({(valid_games / total_games * 100):.2f}%)")


--- ИТОГОВАЯ СТАТИСТИКА ---
Всего игр в датафрейме: 12980
Игр с оценками критиков: 7267 (55.99%)
Игр с корректными оценками (0–100): 7267 (55.99%)


Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [41]:
platform_counts = df_actual['Platform'].value_counts().head(7)


In [42]:
print("=== ТОП-7 ПЛАТФОРМ ПО КОЛИЧЕСТВУ ИГР ===")
print(f"{'№':<3} {'Платформа':<15} {'Количество игр':<15} {'Доля от общего числа (%)':<20}")
print("-" * 55)

total_games_with_platform = df_actual['Platform'].notna().sum()

for i, (platform, count) in enumerate(platform_counts.items(), start=1):
    percentage = (count / total_games_with_platform) * 100
    print(f"{i:<3} {platform:<15} {count:<15} {percentage:<20.2f}")


=== ТОП-7 ПЛАТФОРМ ПО КОЛИЧЕСТВУ ИГР ===
№   Платформа       Количество игр  Доля от общего числа (%)
-------------------------------------------------------
1   PS2             2154            16.59               
2   DS              2146            16.53               
3   WII             1294            9.97                
4   PSP             1199            9.24                
5   XBOX 360        1138            8.77                
6   PLAYSTATION 3   1107            8.53                
7   GBA             826             6.36                


## 5. Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

**Анализ данных датасета /datasets/new_games.csv показал, что распределение игр по оценкам критиков крайне неравномерно:**
* Средняя оценка — самая массовая категория (5500 игр, 42,37%).
* Высокая оценка — вторая по численности (1712 игр, 13,19%).
* Низкая оценка — крайне редкая (55 игр, всего 0,42%).
Топ‑7 платформ по количеству выпущенных игр демонстрирует явное доминирование консолей прошлых поколений (например, PS2, X360, PS3, Wii) и портативных устройств (DS). Это отражает исторический пик игровой индустрии в 2000‑х и начале 2010‑х годов.

   **Анализ распределения игр по категориям оценок критиков:**
- Сгруппированы данные по полю Critic Score Category.
- Подсчитано количество игр в каждой категории (Высокая оценка, Средняя оценка, Низкая оценка).
- Рассчитаны доли каждой категории в процентах от общего числа игр с оценками критиков.
- Выведена итоговая статистика: общее количество игр, число игр с оценками критиков и с корректными оценками (0–100).
-  **Выделение топ‑7 платформ:**
- Выполнена группировка данных по полю Platform.
- Для каждой платформы подсчитано количество связанных с ней игр.
- Отсортированы платформы по убыванию количества игр.
- Отобраны первые 7 позиций — сформирован топ‑7.
-  Для каждой платформы в топе рассчитана доля от общего числа записей с указанной платформой (в процентах).

   **Проверка качества данных:**
- Проанализированы пропущенные значения в поле Platform.
- Учтены записи без оценок критиков при расчёте итоговой статистики.



### Новые поля, добавленные в набор данных

`category_counts` — количество игр по категориям оценок критиков;

`category_percentages` — доли категорий оценок критиков в процентах;

`platform_counts` — топ‑7 платформ с количеством игр;

`total_games`, `games_with_scores`, `valid_games` — итоговая статистика по объёму данных.